pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -441,13 +441,23 @@ func (r *rbac) createResources(ctx context.Context) error {
 			logger.Infof("namespace %s sucessfully reconciled. Adding label namespace-reconcile-version to mark it as reconciled", ns.Name)
 			// Add `openshift-pipelines.tekton.dev/namespace-reconcile-version` label to namespace
 			// so that rbac won't loop on it again
-			nsLabels := ns.GetLabels()
+
+			// Re-fetch the namespace to ensure we have the latest version
+			updatedNS, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, ns.Name, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to re-fetch namespace %s: %v", ns.Name, err)
+			}
+
+			// Get the current labels, or initialize if none exist
+			nsLabels := updatedNS.GetLabels()
 			if len(nsLabels) == 0 {
```

Comment:
no need to check len(nsLabels), check directly if nsLabels == nil , GetLabels will return nil of there are no labels
if nsLabels == nil {
    nsLabels = make(map[string]string)
}

Reply:
Done